### ISSUE TITLE
TimeDeltaParameter can't be passed to dynamic dependencies

### ISSUE THREAD

**Issue by kierkegaard13** [spotify/luigi#1966]
I have a task with a TimeDeltaParameter, let's call it TaskA. I can run TaskA from the command line like so
```
luigi --module project.modules TaskA --buffer '1 minute'
```
TaskA has a dynamic dependency, TaskB, which I yield from the run function like so
```
def run():
    yield TaskB(self.buffer)
```
This results in the error
```
luigi.parameter.ParameterException: Invalid time delta - could not parse 0:01:00
```
It seems like passing TimeDeltaParameters to dynamic dependencies results in this issue, because they are passed as strings and then can't be parsed. Even more interesting, is when requiring TaskB from the requires function, it works just fine. Digging deeper, it looks like the parse function which throws the above error, isn't called when running dependencies from the requires function. Does anyone know how to get around this? I'm also curious as to if there's any way to set a default TimeDeltaParameter of 0 days as that seems to throw the same error as above.

**Comment by Tarrasch** [comment]
I think TimeDeltaParameters were created a long time ago, probably before good coding practices were applied. I think this can be fixed by fixing the `serialize`/`parse` pair. Please take a look @kierkegaard13! :)

**Comment by dlstadther** [comment]
resolved by #1968
